### DISCUSSION TITLE
Feature to get response with request id

### DISCUSSION THREAD

**Discussion by jshi361** [deepgram/community#32] [General help] [answered]
Is there an way to get response back if the callback failed

**Comment by briancbarrow** [comment]
@jshi361 are you using the live streaming or the pre-recorded feature? And are you using one of the SDKs to make the call?

We are working on making this available in the next week or two.

**Reply by jshi361** [reply]
Hi Brian, I am using prerecorded feature and also using the sdk to make the call

**Reply by briancbarrow** [reply]
we hope to have this added soon. I'll post here when we release it.

**Reply by prayashm** [reply]
@briancbarrow any update on this?

**Reply by briancbarrow** [reply]
@prayashm @jshi361 Sorry for the very long delay. This is now released. If you get a error from the API you will have the request_id in the response. The Node SDK requires version 1.18.3 to get that full response. Let me know if you run into more issues.

**Reply by prayashm** [reply]
@briancbarrow Thanks for the reply.
I am using the APIs (not any of the SDKs)

The problem is Deepgram service is not calling the callback URL specified in the request.
I have noticed this happens for long transcripts - it failed for a 20-minute and another 50-minute audio but works fine for shorter audios.

I have the `request ID` of the request submitted to the `v1/listen` endpoint.
Using this in the Dashboard > Usage > Logs - gives me that there was a `200 OK` response for the `POST /listen` call.
I don't know what to do next or where to debug the reason for the failure of the callback URL being called.

Making the synchronous (by removing the callback URL) works fine as the Deepgram API responds in under 2 minutes max - as a workaround. Ideally, I would want to keep this async.

**Reply by rilhia** [reply]
@briancbarrow I've experienced precisely what @prayashm has described.

**Reply by prayashm** [reply]
Update: Got on a call with support - they checked - and it was a Body size limit causing the callbacks to fail with `413 Status code` from our webserver. Updating the incoming payload size limits would fix it. This makes sense as longer transcripts will have more text.

In our case, the default limit was 1 MB on nginx.

**Reply by rilhia** [reply]
Thanks for this info @prayashm. It turns out that I am experiencing the same cause. The payload limit is 512KB. Since the error status will be sent back to Deepgram, it would make sense to be able to record these. But I agree that being able specify a max payload size for callbacks would be really useful.

**Comment by BekahHW** [comment]
@jshi361 Thanks for asking! Next time you have a question, put it in the Q&A category in a new post to make sure it gets answered 🙂

**Comment by lukeocodes** [comment, marked as answer]
This was resolved by following the steps in this guide: https://developers.deepgram.com/docs/payload-too-large

The requests were hitting a Body Size Limit error, where their callback server was rejecting the transcript due to the size of our callback request.
